aws/client/logger.go

```diff
@@ -94,6 +94,10 @@ var LogHTTPRequestHeaderHandler = request.NamedHandler{
 }
 
 func logRequestHeader(r *request.Request) {
+	if !r.Config.LogLevel.AtLeast(aws.LogDebug) {
```

Comment:
Should also check if `r.Config.Logger` is nil?

The `logResponseHeader` function also needs an update adding a check for Loglevel to atleast match `aws.LogDebug`

Reply:
Also `logRequest` and `logResponse` assumes `r.Config.Logger ` is not nil.  Do we validate this somewhere or explicitly set a default that cannot be removed?